### ISSUE TITLE
Keyboard navigation not working in Select component

### ISSUE THREAD

**Issue by stropitek** [zakodium-oss/react-science#500]
Might be related to shadow dom

See conversation in https://github.com/zakodium-oss/react-science/pull/495

**Comment by stropitek** [comment]
We removed both the Select component and the shadow dom.

But the story for Select still does not have keyboard navigation. The blueprintjs component supports keyboard navigation so this seems surprising.

Can you debug this @moonayyur ?

**Comment by moonayyur** [comment]
@stropitek keyboard navigation works when adding OnActiveItemChange in `Select`, but it triggers an update loop with the first element of the array :

Egs: when selecting Orange :
![image](https://github.com/zakodium-oss/react-science/assets/74436107/05ad4480-fd99-4dcb-b2b2-7cf8fa716837)

The keyboard functionality can be tested using this code (excluding the first element) : 

```
onActiveItemChange={(item) => {
          if (item?.label !== 'Apple') {
            setValue(item);
            console.log('Item :', item);
          }
        }}
```

**Comment by stropitek** [comment]
@moonayyur the issue was that the active element was controlled when it shouldn't
